test/e2e/pods.go

```diff
@@ -555,6 +578,233 @@ var _ = Describe("Pods", func() {
 		}, false)
 	})
 
+	It("should reset back-off on image update", func() {
+		name := "backoff"
+		containerName := "back-off"
+		podClient := framework.Client.Pods(framework.Namespace.Name)
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name:   name,
+				Labels: map[string]string{"test": "backoff-image"},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    containerName,
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"/bin/sh", "-c", "sleep 1", "/crash/missing"},
+					},
+				},
+			},
+		}
+
+		By("submitting the pod to kubernetes")
+		defer func() {
+			By("deleting the pod")
+			podClient.Delete(pod.Name, api.NewDeleteOptions(0))
+		}()
+		pod, err := podClient.Create(pod)
+		if err != nil {
+			Failf("Failed to create pod: %v", err)
+		}
+
+		expectNoError(framework.WaitForPodRunning(pod.Name))
+
+		By("verifying the pod is in kubernetes")
+		pod, err = podClient.Get(name)
+		if err != nil {
+			Failf("failed to get pod: %v", err)
+		}
+
+		By("Build backoff offset")
+		time.Sleep(time.Minute) // build a restart gap
+
+		By("geting restart delay-1")
+		delay1, err := getRestartDelay(framework.Client, pod, framework.Namespace.Name, name, containerName)
+		if err != nil {
+			Failf("timed out waiting for container restart in pod=%s/%s", name, containerName)
+		}
+
+		By("getting restart delay-2")
+		delay2, err := getRestartDelay(framework.Client, pod, framework.Namespace.Name, name, containerName)
+		if err != nil {
+			Failf("timed out waiting for container restart in pod=%s/%s", name, containerName)
+		}
+
+		ratio := float64(delay2) / float64(delay1)
+		if math.Floor(ratio) != 2 && math.Ceil(ratio) != 2 {
+			Failf("back-off gap is not increasing exponentially pod=%s/%s delay1=%s delay2=%s", name, containerName, delay1, delay2)
+		}
+
+		By("verifying the pod is in kubernetes")
+		pod, err = podClient.Get(name)
+		if err != nil {
+			Failf("failed to get pod: %v", err)
+		}
+
+		By("updating the image")
+		pod.Spec.Containers[0].Image = "nginx"
+		pod, err = podClient.Update(pod)
+		if err != nil {
+			Failf("error updating pod=%s/%s %v", name, containerName, err)
+		}
+		time.Sleep(10 * time.Second)
+		expectNoError(framework.WaitForPodRunning(pod.Name))
+
+		By("get restart delay after image update")
+		delayAfterUpdate, err := getRestartDelay(framework.Client, pod, framework.Namespace.Name, name, containerName)
+		if err != nil {
+			Failf("timed out waiting for container restart in pod=%s/%s", name, containerName)
+		}
+
+		if delayAfterUpdate > delay2 || delayAfterUpdate > delay1 {
+			Failf("updating image did not reset the back-off value in pod=%s/%s d3=%s d2=%s d1=%s", name, containerName, delayAfterUpdate, delay1, delay2)
+		}
+
+	})
+
+	It("should not back-off on LivenessProbe failure", func() {
```

Comment:
The code in this test is largely a cut-n-paste of the code in the previous one (which in turn contains a lot of cut-n-paste code from earlier tests).  Please refactor to remove duplicate code.